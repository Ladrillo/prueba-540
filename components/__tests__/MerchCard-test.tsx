import { render } from '@testing-library/react-native'
import MerchantCard from '../MerchCard'

const order1 = {
  id: '1',
  merchantName: 'Amazon',
  merchantImage: 'https://res.cloudinary.com/caskchain/image/upload/v1717158058/Sequra/amazon.png',
  merchantLogo: 'https://cdn.iconscout.com/icon/free/png-512/free-amazon-1543560-1306063.png',
  date: '2021-10-01',
  nextDueAmount: 100,
  nextDueDate: '2021-11-01',
  status: 'pending',
  reference: '123456789',
  price: 100,
  numberOfArticles: 5,
  shippedArticles: 2,
}

describe('<MerchantCard />', () => {
  test('Text renders correctly on MerchantCard', () => {
    const screen = render(<MerchantCard data={order1} handlePress={Function.prototype} />)
    screen.getByText('100 â‚¬')
    expect(screen.toJSON()).toMatchSnapshot()
  })
})
