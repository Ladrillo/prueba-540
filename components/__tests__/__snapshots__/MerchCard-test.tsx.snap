// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MerchCard /> Text renders correctly on MerchCard 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 15,
        "elevation": 5,
        "margin": 20,
        "marginBottom": 0,
        "overflow": "hidden",
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 5,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        {
          "height": 120,
          "justifyContent": "space-between",
          "resizeMode": "cover",
          "width": "100%",
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://res.cloudinary.com/etc.png",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": 120,
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 15,
          }
        }
      >
        <View>
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                {
                  "color": "#fff",
                  "fontSize": 20,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Amazon
          </Text>
          <View
            style={
              {
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  {
                    "color": "#fff",
                    "marginRight": 10,
                  },
                ]
              }
            >
              1 nov 2021
            </Text>
            <Text
              style={
                [
                  {
                    "color": "#11181C",
                  },
                  {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  {
                    "color": "#fff",
                  },
                ]
              }
            >
              <Text />
               
              5
               artículos
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "marginRight": 10,
              "width": 40,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://cdn.iconscout.com/etc.png",
              }
            }
            style={
              {
                "borderRadius": 17.5,
                "height": 35,
                "width": 35,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 15,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#e0f7fa",
            "borderRadius": 10,
            "height": 110,
            "justifyContent": "center",
            "paddingHorizontal": 15,
            "paddingVertical": 10,
            "width": 110,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          1
        </Text>
        <Text
          style={
            {
              "fontSize": 14,
            }
          }
        >
          noviembre
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "flex-end",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "color": "#7e7e7e",
                "fontSize": 14,
              },
            ]
          }
        >
          Cobro de:
        </Text>
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "color": "#000",
                "fontSize": 22,
                "fontWeight": "bold",
              },
            ]
          }
        >
          100
           €
        </Text>
      </View>
    </View>
  </View>
</View>
`;
